sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala

```diff
@@ -3631,6 +3631,29 @@ private[sql] object QueryCompilationErrors extends QueryErrorsBase with Compilat
     )
   }
 
+  def implicitCollationMismatchError(): Throwable = {
+    new AnalysisException(
+      errorClass = "COLLATION_MISMATCH.IMPLICIT",
+      messageParameters = Map.empty
+    )
+  }
+
+  def explicitCollationMismatchError(explicitTypes: Seq[String]): Throwable = {
+    new AnalysisException(
+      errorClass = "COLLATION_MISMATCH.EXPLICIT",
+      messageParameters = Map(
+        "explicitTypes" -> toSQLId(explicitTypes)
+      )
+    )
+  }
+
+  def indeterminateCollationError(): Throwable = {
```

Comment:
Is it used somewhere or supposed to use? BTW, this PR https://github.com/apache/spark/pull/47740 is going to remove it.